storage/interface.go

```diff
@@ -0,0 +1,86 @@
+package storage
+
+import (
+	"errors"
+	"github.com/nuts-foundation/nuts-node/core"
+)
+
+// ErrCommitFailed is returned when the commit of transaction fails.
+var ErrCommitFailed = errors.New("unable to commit transaction")
+
+// Engine defines the interface for the storage engine.
+type Engine interface {
+	core.Engine
+	core.Configurable
+	core.Runnable
+	Warehouse
+}
+
+// Warehouse is used to manage data stores.
+type Warehouse interface {
+	// GetKVStore returns a key-value store for the given engine and store name.
+	GetKVStore(engineKey string, storeName string) (KVStore, error)
+}
+
+// KVStore defines the interface for a key-value store.
+// Writing to it is done in callbacks passed to the Write-functions. If the callback returns an error, the transaction is rolled back.
+type KVStore interface {
+	// Close releases all resources associated with the KVStore. It is safe to call multiple (subsequent) times.
+	Close() error
+	// Write starts a writable transaction and passes it to the given function.
+	Write(fn func(WriteTransaction) error) error
+	// Read starts a read-only transaction and passes it to the given function.
+	Read(fn func(transaction ReadTransaction) error) error
+	// WriteBucket starts a writable transaction, retrieves the specified bucket and passes it to the given function.
+	// If the bucket does not exist, it will be created.
+	WriteBucket(bucketName string, fn func(BucketWriter) error) error
+	// ReadBucket starts a read-only transaction, retrieves the specified bucket and passes it to the given function.
+	// If the bucket does not exist, the function is not called.
+	ReadBucket(bucketName string, fn func(BucketReader) error) error
+}
+
+// WriteTransaction is used to write to a KVStore.
+type WriteTransaction interface {
```

Comment:
`OnRollback` and `OnCommit` are required here

Reply:
I added them as AfterRollback and AfterCommit to make it clear it's really exactly that (and not before commit/rollback)